### PR TITLE
Use dashes in Azure and S3 options for consistency

### PR DESCRIPTION
**Pull request by julianohubel**
Addressed to issue [#3748](https://github.com/duplicati/duplicati/issues/3748#issue-436683140)

### CODE REVIEW

File: Duplicati/Library/Backend/S3/S3Backend.cs [hunk 1]
```diff
@@ -392,7 +392,9 @@ public IList<ICommandLineArgument> SupportedCommands
 
                 var normal = new ICommandLineArgument[] {
                     new CommandLineArgument("aws_secret_access_key", CommandLineArgument.ArgumentType.Password, Strings.S3Backend.AMZKeyDescriptionShort, Strings.S3Backend.AMZKeyDescriptionLong, null, new string[] {"auth-password"}, null),
```
Comment (warwickmm) [inline Duplicati/Library/Backend/S3/S3Backend.cs:394]: For the old underscored options, can we use the following `CommandLineArgument` constructor that marks the options as deprecated?

https://github.com/duplicati/duplicati/blob/67aaab15e3a8638806a50cf23b5a085c826a338a/Duplicati/Library/Interface/CommandLineArgument.cs#L258-L263

Comment (julianohubel) [reply Duplicati/Library/Backend/S3/S3Backend.cs]: Shure. I added a new commit change. Could you review?

File: Duplicati/Library/Backend/AzureBlob/AzureBlobBackend.cs [hunk 2]
```diff
@@ -131,15 +131,18 @@ public IList<ICommandLineArgument> SupportedCommands
                     new CommandLineArgument("azure_account_name",
                         CommandLineArgument.ArgumentType.String,
                         Strings.AzureBlobBackend.StorageAccountNameDescriptionShort,
-                        Strings.AzureBlobBackend.StorageAccountNameDescriptionLong),
+                        Strings.AzureBlobBackend.StorageAccountNameDescriptionLong,
+                        null, null, null, "This is deprecated, use azure-account-name insted"),
```
Comment (warwickmm) [inline Duplicati/Library/Backend/AzureBlob/AzureBlobBackend.cs:135]: Thanks.  Can we change the spelling to `instead`?

Comment (julianohubel) [reply Duplicati/Library/Backend/AzureBlob/AzureBlobBackend.cs]: Right. I Added another commit, thanks for the correction.

**Comment by warwickmm** [comment]
Thanks @julianohubel.  I can see the deprecated messages for azure:
```
$ mono Duplicati/GUI/Duplicati.GUI.TrayIcon/bin/Debug/Duplicati.CommandLine.exe help azure
Azure blob (azure):
 This backend can read and write data to Azure blob storage.  Allowed formats are: "azure://bucketname"
 Supported options:
  --azure_account_name (String): The storage account name
    [DEPRECATED]: This is deprecated, use azure-account-name instead
    The Azure storage account name which can be obtained by clicking the "Manage Access Keys" button on the storage account dashboard
  --azure_access_key (Password): The access key
    [DEPRECATED]: This is deprecated, use azure-access-key instead
    The Azure access key which can be obtained by clicking the "Manage Access Keys" button on the storage account dashboard
  --azure_blob_container_name (String): The name of the storage container
    [DEPRECATED]: This is deprecated, use azure-blob-container-name instead
    All files will be written to the container specified
  --azure-account-name (String): The storage account name
    The Azure storage account name which can be obtained by clicking the "Manage Access Keys" button on the storage account dashboard
  --azure-access-key (Password): The access key
    The Azure access key which can be obtained by clicking the "Manage Access Keys" button on the storage account dashboard
  --azure-blob-container-name (String): The name of the storage container
    All files will be written to the container specified
  --auth-password (Password): Supplies the password used to connect to the server
    The password used to connect to the server. This may also be supplied as the environment variable "AUTH_PASSWORD".
  --auth-username (String): Supplies the username used to connect to the server
    The username used to connect to the server. This may also be supplied as the environment variable "AUTH_USERNAME".
```
However, they don't appear for s3:
```
$ mono Duplicati/GUI/Duplicati.GUI.TrayIcon/bin/Debug/Duplicati.CommandLine.exe help s3

Duplicati can use S3-compatible servers to store backups. The following target URL format is used:
  s3://bucketname/prefix

Options:
  --aws_secret_access_key
    The AWS "Secret Access Key" can be obtained after logging into your AWS account, this can also be supplied through the "auth-password" property
  --aws-secret-access-key
    The AWS "Secret Access Key" can be obtained after logging into your AWS account, this can also be supplied through the "auth-password" property
  --aws_access_key_id
    The AWS "Access Key ID" can be obtained after logging into your AWS account, this can also be supplied through the "auth-username" property
  --aws-access-key-id
    The AWS "Access Key ID" can be obtained after logging into your AWS account, this can also be supplied through the "auth-username" property
...
```
I don't quite understand the difference, but it appears that for `azure`, the following is `false`, while for `s3` it is `true`:
https://github.com/duplicati/duplicati/blob/c47a356d6f2fafbcb990957bcd364df7fe16a80c/Duplicati/CommandLine/Help.cs#L92

For `s3`, this causes the output to be displayed using `PrintArgSimple`, which doesn't respect the deprecated property:
https://github.com/duplicati/duplicati/blob/c47a356d6f2fafbcb990957bcd364df7fe16a80c/Duplicati/CommandLine/Help.cs#L486-L496

I'm not that familiar with this part of the code, so I'm not sure what the best way to proceed is right now.  Any ideas?  I don't know why there are so many different code paths for printing options.

**Comment by warwickmm** [comment]
I think the difference is s3 has an entry here:

https://github.com/duplicati/duplicati/blob/85c9e4394308933d549face3b9cf97ec37c13978/Duplicati/CommandLine/help.txt#L382-L389

Since the following is `true`

https://github.com/duplicati/duplicati/blob/85c9e4394308933d549face3b9cf97ec37c13978/Duplicati/CommandLine/Help.cs#L188

we then replace `%MODULEOPTIONS%` with the result from `PrintArgsSimple`, which doesn't respect the deprecated flag

https://github.com/duplicati/duplicati/blob/85c9e4394308933d549face3b9cf97ec37c13978/Duplicati/CommandLine/Help.cs#L229-L230

One minimally invasive option is to modify `PrintArgsSimple` to call `CommandLineArgument.PrintArgument`, which seems to fix the s3 case:
```
diff --git a/Duplicati/CommandLine/Help.cs b/Duplicati/CommandLine/Help.cs
index 843396fc6..66cf9ba55 100644
--- a/Duplicati/CommandLine/Help.cs
+++ b/Duplicati/CommandLine/Help.cs
@@ -489,8 +489,10 @@ namespace Duplicati.CommandLine
                 return "";
 
             List<string> lines = new List<string>();
-            foreach(Duplicati.Library.Interface.ICommandLineArgument arg in args)
-                lines.Add(PrintArgSimple(arg, arg.Name));
+            foreach (Duplicati.Library.Interface.ICommandLineArgument arg in args)
+            {
+                Duplicati.Library.Interface.CommandLineArgument.PrintArgument(lines, arg);
+            }
 
             return string.Join(Environment.NewLine, lines.ToArray());
         }
```

@julianohubel, what do you think?

**Comment by kenkendk** [comment]
The PR suggested by @julianohubel is correct. The render code is broken.

I think the idea is that we wanted to sometimes show a limited set of options.
I will look an try to fix the help rendering mess.

**Comment by kenkendk** [comment]
It is still a bit of a mess, but the idea is:

1a) If there is an entry in `help.txt` that has been created by a human who decided how it should look.
1b) If the entry in `help.txt` contains "%MODULEOPTIONS%", print the module options in a concise manner

2a) If there are no entries in `help.txt`, auto-create one by using the description from the module.
2b) Print each option with the same code that is used for printing help for a specific option.

It would perhaps make sense to just append the same options list in either case.
I did not change that, but I did fix the output in #4459

**Comment by ts678** [comment]
[AWS Secret Access Key Warning in Duplicati - 2.0.5.113_canary_2021-03-07](https://forum.duplicati.com/t/aws-secret-access-key-warning-in-duplicati-2-0-5-113-canary-2021-03-07/11901) seems related to this. Is issue likely to be widespread?

On a side note, is the UI still using the old option?

https://github.com/duplicati/duplicati/blob/f145b9fcc2f93aaa606bc939373afbb566034647/Duplicati/Server/webroot/ngax/scripts/services/EditUriBuiltins.js#L341-L345

**Comment by warwickmm** [comment]
I can't reproduce the warnings.  There is an outstanding issue (#3082) regarding old options hanging around after changing backends.  Could that be the cause?

I created issue #4461 to fix the UI and will submit a fix soon.  Thanks for noticing that.

**Comment by warwickmm** [comment]
I just noticed that the user in the forum says they're using the Wasabi backend.  I believe this is an S3-compatible backend, so that would make sense why they're seeing warnings about deprecated S3 options.

**Pull request merged**